File: src/Sequence/SqBundle.java
package Sequence;

import static arc.Core.bundle;

public class SqBundle {
    private static final StringBuilder builder = new StringBuilder();

    public static String format(String key, Object ...args) {
        return bundle.format(key, args);
    }

    public static String get(String key, String ...def) {
        return def.length > 0 ? bundle.get(key, def[0]) : bundle.get(key);
    }

    public static String cat(String ...args) {
        builder.setLength(0);
        for (String s : args) {
            builder.append(s).append('.');
        }
        builder.deleteCharAt(builder.length() - 1);
        return builder.toString();
    }

    public static String modCat(String ...args) {
        return get(SeqMod.MOD.meta.name + "." + cat(args));
    }

    public static String catGet(String ...args) {
        return get(cat(args));
    }
}

File: src/sequence/core/SqBundle.kt
package sequence.core

import arc.Core
import sequence.SeqMod

object SqBundle {
    private val builder = StringBuilder()
    fun format(key: String, vararg args: Any?): String {
        return Core.bundle.format(key, *args)
    }

    operator fun get(key: String, def: String): String = Core.bundle[key, def]

    operator fun invoke(key: String, def: String): String = get(key, def)

    operator fun get(key: String): String = Core.bundle[key]

    operator fun invoke(key: String): String = get(key)

    fun cat(vararg args: String): String {
        builder.setLength(0)
        for (s in args) {
            builder.append(s).append('.')
        }
        builder.deleteCharAt(builder.length - 1)
        return builder.toString()
    }

    fun mod(arg: String): String {
        return SqBundle[SeqMod.MOD.meta.name + "." + arg]
    }

    fun catGet(vararg args: String): String {
        return SqBundle[cat(*args)]
    }
}
